app/commands.go

```diff
@@ -615,15 +614,33 @@ type searchCmd struct {
 	Constraint string `arg:"" help:"Package regex." optional:""`
 }
 
-func (s *searchCmd) Run(l *ui.UI, env *hermit.Env) error {
-	err := env.Sync(l, false)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	pkgs, err := env.Search(l, s.Constraint)
-	if err != nil {
-		return errors.WithStack(err)
+func (s *searchCmd) Run(l *ui.UI, env *hermit.Env, state *state.State) error {
+	var (
+		pkgs manifest.Packages
+		err  error
+	)
+	if env != nil {
+		err = env.Sync(l, false)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		pkgs, err = env.Search(l, s.Constraint)
```

Comment:
What's the difference between `env.Search` and `state.Search`?

Reply:
env.Search() will search sources from the environment config.

Reply:
```go
type Searchable {
  Sync(*ui.UI, bool) error
  Search(*ui.UI, Constraint) Packages[], error
}
```

Or something. I can't write Go in a textbox yet.

Reply:
That did cross my mind but I decided YAGNI.

Reply:
In general I usually add abstractions if there are greater than 2-3 usages.